helper/src/main/java/me/lucko/helper/utils/XMaterial.java

```diff
@@ -0,0 +1,1686 @@
+        // So you can still use this name even if it's a duplicated name.
+        // Since duplicated names only apply to older versions.
+        if (newMat != null && (isNewVersion() || !this.isDuplicated())) return newMat;
+        return requestOldMaterial(suggest);
+    }
+
+    /**
+     * Parses from old material names and can accept suggestions.
+     *
+     * @param suggest Accept suggestions for newly added blocks
+     * @return A parsed Material suitable for this minecraft version.
+     */
+    @Nullable
+    private Material requestOldMaterial(boolean suggest) {
+        Material oldMat;
+        boolean isNew = getVersionIfNew() != MinecraftVersion.UNKNOWN;
+        for (int i = this.legacy.length - 1; i >= 0; i--) {
+            String legacyName = this.legacy[i];
+            // Slash means it's just another name for the material in another version.
+            if (legacyName.contains("/")) {
+                oldMat = Material.getMaterial(parseLegacyVersionMaterialName(legacyName));
+
+                if (oldMat != null) return oldMat;
+                else continue;
+            }
+            if (isNew) {
+                if (suggest) {
+                    oldMat = Material.getMaterial(legacyName);
+                    if (oldMat != null) return oldMat;
+                } else return null;
+                // According to the suggestion format list, all the other names continuing
+                // from here are considered as a "suggestion" if there's no slash anymore.
+            }
+            oldMat = Material.getMaterial(legacyName);
+            if (oldMat != null) return oldMat;
+        }
+        return null;
+    }
+
+    /**
+     * Checks if an item is similar to the material and its data (if in older versions.)
+     *
+     * @param item item to check.
+     * @return true if the material is the same as the item's material (and data if in older versions.)
+     */
+    @SuppressWarnings("deprecation")
+    public boolean isSimilar(@Nonnull ItemStack item) {
+        Objects.requireNonNull(item, "ItemStack cannot be null");
+        Objects.requireNonNull(item.getType(), "ItemStack's material cannot be null");
+        return (isNewVersion() || this.isDamageable()) ? item.getType() == this.parseMaterial() : item.getType() == this.parseMaterial() && item.getDurability() == this.data;
+    }
+
+    /**
+     * Get the suggested material names that can be used instead of this material.
+     *
+     * @return a list of suggested material names.
+     */
+    @Nonnull
+    public String[] getSuggestions() {
+        if (this.legacy.length == 0 || !this.legacy[0].contains(".")) return new String[0];
+        return Arrays.stream(this.legacy).filter(mat -> !mat.contains(".")).toArray(String[]::new);
+    }
+
+    /**
+     * Checks if this material is supported in the current version.<br>
+     * It'll check both the newest matetrial name and for legacy names.
```

Comment:
Additional paragraphs in comments should include an empty line between previous paragraphs.